### PR TITLE
Instrument RunnableFuture and RejectedExecutionHandler

### PR DESCRIPTION
**Pull request by richardstartin**
* Removes the deprecation of `AgentTracer.activeScope` - we use this method a lot and there is no other way to do what it does.
* Removes all `ExecutorService` instrumentation for API not inherited from `Executor`
* Removes `ScheduledExecutorService` instrumentation
* Prevents the need to instrument `Callable` or `Future` at all
* Not quite enough to prevent from instrumenting `Runnable` and `Executor` - this will need wrapping and I haven't got that working yet - so this won't dramatically improve our clojure story yet.
* Adds more testing for executor services in general, adding test dependencies on some of the libraries we instrument executors from into `java-concurrent` 
* instruments `RejectedExecutionHandler` to capture rejected work which doesn't result in an exception being thrown. Because we're now lenient about continuation reference counts, this can't cause any problems, but this allows rejected execution to be handled in a timely fashion.

Rebased on top of #1972

### CODE REVIEW

File: dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class SwallowingRejectedExecutionHandler implements RejectedExecutionHandler {
+  @Override
+  public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+    System.out.println(executor + " rejected " + r);
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java:7]: `println`?

**Review by bantonsson — APPROVED** [review]

File: dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java (hunk 1 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java:7]: It's Java

Comment (bantonsson) [reply dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java:7]: Sorry. I was wondering why there is a `System.out.println(...)` there at all.

Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java:7]: When tests fail it's nice to have some context, this seems relevant to me. I can change it to be a log statement if you like.

Comment (bantonsson) [reply dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java:7]: Ok, just leave it there.

File: dd-java-agent/instrumentation/java-concurrent/src/test/groovy/ForkJoinPoolPropagationTest.groovy [hunk 2]
```diff
@@ -4,7 +4,7 @@ import datadog.trace.core.DDSpan
 import java.util.concurrent.ForkJoinPool
 
 class ForkJoinPoolPropagationTest extends AgentTestRunner {
-  def "test imbalanced recursive task propagation #parallelism FJP threads" () {
+  def "test imbalanced recursive task propagation #parallelism FJP threads (async #async)" () {
```
Comment (mcculls) [inline dd-java-agent/instrumentation/java-concurrent/src/test/groovy/ForkJoinPoolPropagationTest.groovy:7]: the `async` variable only appears to be used in the test name - is this right?

Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/test/groovy/ForkJoinPoolPropagationTest.groovy:7]: good catch, it was supposed to go in to the FJP constructor...

**Review by mcculls — APPROVED** [review]
+1 (reviewing last 2 commits)

File: dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java (hunk 1 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/test/java/SwallowingRejectedExecutionHandler.java]: I removed it.

**Pull request merged**